.github/workflows/rspec_rubocop.yml

```diff
@@ -0,0 +1,77 @@
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
+
+on: [pull_request, push]
+
+permissions:
+  contents: read
+
+env:
+  CC: ccache gcc
+  CXX: ccache g++
+  THRIFT_SHA256SUM: 2289d02de6e8db04cbbabb921aeb62bfe3098c4c83f36eec6c31194301efa10b  thrift-0.10.0.tar.gz
+  THRIFT_URL: http://archive.apache.org/dist/thrift/0.10.0/thrift-0.10.0.tar.gz
+  THRIFT_FILE: thrift-0.10.0.tar.gz
+  THRIFT_DIR: thrift-0.10.0
+  DEBIAN_FRONTEND: noninteractive # for apt to be noninteractive
+  ImageOS: ubuntu20
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: ubuntu:focal
```

Comment:
The container used by GitHub actions comes with a lot of preinstalled packages – see [this](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md). One of those packages is Haskell – and its respective tools – which causes an error when building thrift:

```
Updated package list of hackage.haskell.org to the index-state 2023-02-22T13:05:59Z
cabal: Could not resolve dependencies:

...

After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, vector, primitive, thrift
Try running with --minimize-conflict-set to improve the error message.
```

Seems that it is an error on Thrift v0.10.0 – see this [report](https://hackage.haskell.org/package/thrift-0.10.0/reports/2).

In order to get around this, I set up GitHub Actions so that it uses Docker's official Ubuntu Focal image instead.

Reply:
Another option could be to use Github Actions's container but removing `Haskell` (`rm -rf /opt/ghc`?). Although I didn't tested it 🤔

Reply:
I agree that starting from the official ubuntu image is a better experience than trying to undo or remove features that come pre-installed on the github action container 👍